File: frontend/src/types/crawlCookie.ts
/**
 * Cookie 库相关的类型定义。
 *
 * 字段命名与后端 API 保持一致（snake_case），不做 camelCase 转换。
 */

import type { CrawlPlatform } from './crawler'

/** 一条保存的 Cookie（详情接口返回，含完整串） */
export interface CrawlCookie {
  id: number
  platform: CrawlPlatform
  platform_label: string
  name: string
  cookie: string
  remark: string
  created_at: string
  updated_at: string
}

/** 列表项（只给截断预览，不给完整 cookie 串） */
export interface CrawlCookieListItem {
  id: number
  platform: CrawlPlatform
  platform_label: string
  name: string
  cookie_preview: string
  remark: string
  created_at: string
  updated_at: string
}

/** 保存 / 更新 Cookie 的请求体 */
export interface CrawlCookiePayload {
  platform: CrawlPlatform
  name: string
  cookie: string
  remark?: string
}

/** 列表接口返回（量少不分页） */
export interface CrawlCookieListData {
  total: number
  items: CrawlCookieListItem[]
}

File: backend/app/models/crawl_cookie.py
"""Cookie 库 ORM 模型。

素材抓取「Cookie 登录」的凭据库：按平台保存常用的登录 Cookie，
抓取时从库里选取，免去每次手工粘贴。

与 CrawlJob.login_cookies 的区别：那里是单次任务的快照（用完即弃），
这里是用户主动维护的可复用凭据。
"""

from datetime import datetime

from sqlalchemy import DateTime, Index, Integer, String, Text, UniqueConstraint
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base
from app.models.content import utcnow


class CrawlCookie(Base):
    """Cookie 库表。"""

    __tablename__ = "crawl_cookies"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True, comment="主键")
    platform: Mapped[str] = mapped_column(
        String(20), nullable=False, comment="平台标识（CrawlPlatform.ALL）"
    )
    name: Mapped[str] = mapped_column(
        String(50), nullable=False, comment="展示名（如「主号」「小号」）"
    )
    cookie: Mapped[str] = mapped_column(
        Text, nullable=False, comment="登录 Cookie 串（敏感凭据，仅用于抓取时透传）"
    )
    remark: Mapped[str] = mapped_column(String(200), nullable=False, default="", comment="备注")
    created_at: Mapped[datetime] = mapped_column(
        DateTime, nullable=False, default=utcnow, comment="创建时间（UTC）"
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime, nullable=False, default=utcnow, onupdate=utcnow, comment="更新时间（UTC）"
    )

    __table_args__ = (
        # 同一平台下名称唯一：名字就是用户辨认的标识，重复没有区分意义
        UniqueConstraint("platform", "name", name="uq_crawl_cookies_platform_name"),
        Index("ix_crawl_cookies_platform", "platform"),
    )

    def __repr__(self) -> str:
        return f"<CrawlCookie id={self.id} platform={self.platform!r} name={self.name!r}>"
